packages/flutter_tools/templates/app_shared/linux.tmpl/my_application.cc.tmpl

```diff
@@ -117,6 +117,13 @@ static void my_application_class_init(MyApplicationClass* klass) {
 static void my_application_init(MyApplication* self) {}
 
 MyApplication* my_application_new() {
+  // Set the application ID for Wayland. Without this the only identification
+  // the running application will have is the binary name. This way the app can
+  // identify itself via its reverse domain name, the same as the desktop file,
+  // allowing the desktop environment to match the running instance with the
+  // desktop file.
```

Comment:
The comment feels too Wayland specific, as this is actually used in a number of places. Perhaps:
```
Set the program name to the application ID instead of the default binary name. This is used so GTK and shells can map this application to the .desktop file for this application (which needs to be named using the application ID).
```

Reply:
Specifically we probably don't want to mention reverse domain naming as an application ID doesn't need to meet that (but generally does).

Reply:
Good points! I made another pass at the comment. How does that sound now?

Reply:
Perfect!